main/src/ca/uwaterloo/flix/tools/pkg/ManifestParser.scala

```diff
@@ -195,70 +192,74 @@ object ManifestParser {
     */
-  private def collectDependencies(deps: TomlTable, flixDep: Boolean, prodDep: Boolean, p: Path): Result[List[Dependency], ManifestError] = {
-    val depsEntries = deps.entrySet()
-    val depsSet = mutable.Set.empty[Dependency]
-    depsEntries.forEach(entry => {
-      val depName = entry.getKey
-      val depVer = entry.getValue
-      try {
-        toSemVer(depVer.asInstanceOf[String], p) match {
-          case Ok(version) => if (flixDep) {
-            depName.split(':') match {
-              case Array(repo, rest) =>
-                rest.split('/') match {
-                  case Array(username, projectName) =>
-                    if(repo == "github") {
-                      checkNameCharacters(username, p) match {
-                        case Ok(_) => //the name is fine, do nothing
-                        case Err(e) => return Err(e)
-                      }
-                      checkNameCharacters(projectName, p) match {
-                        case Ok(_) => //the name is fine, do nothing
-                        case Err(e) => return Err(e)
-                      }
-                      if (prodDep) {
-                        depsSet.add(Dependency.FlixDependency(Repository.GitHub, username, projectName, version, DependencyKind.Production))
-                      } else {
-                        depsSet.add(Dependency.FlixDependency(Repository.GitHub, username, projectName, version, DependencyKind.Development))
-                      }
-                    } else {
-                      return Err(ManifestError.UnsupportedRepository(p, s"The repository $repo is not supported"))
+  private def collectDependencies(deps: Option[TomlTable], flixDep: Boolean, prodDep: Boolean, p: Path): Result[List[Dependency], ManifestError] = {
+    deps match {
+      case None => Ok(List.empty)
+      case Some(deps) =>
+        val depsEntries = deps.entrySet()
+        val depsSet = mutable.Set.empty[Dependency]
+        depsEntries.forEach(entry => {
+          val depName = entry.getKey
+          val depVer = entry.getValue
+          try {
+            toSemVer(depVer.asInstanceOf[String], p) match {
+              case Ok(version) => if (flixDep) {
+                depName.split(':') match {
+                  case Array(repo, rest) =>
+                    rest.split('/') match {
+                      case Array(username, projectName) =>
+                        if (repo == "github") {
+                          checkNameCharacters(username, p) match {
+                            case Ok(_) => //the name is fine, do nothing
+                            case Err(e) => return Err(e)
+                          }
+                          checkNameCharacters(projectName, p) match {
+                            case Ok(_) => //the name is fine, do nothing
+                            case Err(e) => return Err(e)
+                          }
+                          if (prodDep) {
```

Comment:
I think the nesting depth and use of early return is getting a bit out of hand. 

We should be writing some small functions. Then we can use combinators like `map` and `traverse`, and monadic for.

Its probably easiest to explain over zoom :)